src/styles/block/calendar.scss

```diff
@@ -0,0 +1,58 @@
+.calendar {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  &__block {
+    display: flex;
+
+    gap: $gap;
+    width: $width;
+    flex-wrap: wrap;
+
+    padding: 50px $calendar-padding 0;
+
+    @each $day, $number in $days-of-week {
+      &--start-day--#{$day} {
+        & :first-child {
+          margin-left: $calendar-day-width * $number + $gap * 6;
+        }
+      }
+    }
+
+    @for $index from 28 through 31 {
+      &--month-length--#{$index} {
+        & :nth-child(1n + #{$index + 1}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  &__day {
+    display: flex;
+
+    text-align: center;
+    justify-content: center;
+    align-items: center;
+
+    background-color: #eee;
+    border: 1px solid black;
+
+    transition: 0.3s;
```

Comment:
Better to define which exact properties will be animated, because if you not define it, it will be applied to every property that can be animated